lib/auth/scopes/joining/service.go

```diff
@@ -70,7 +83,22 @@ func (s *Server) CreateScopedToken(ctx context.Context, req *scopedjoiningv1.Cre
 		return nil, trace.AccessDenied("user %q does not have permission to create scoped tokens", authzContext.User.GetName())
 	}
 
-	return (scopedjoiningv1.UnimplementedScopedJoiningServiceServer{}).CreateScopedToken(ctx, req)
+	token := req.GetToken()
+	if token.GetMetadata().GetName() == "" {
+		if token.Metadata == nil {
+			token.Metadata = &headerv1.Metadata{
+				Namespace: apidefaults.Namespace,
```

Comment:
@fspmarshall do we need to set the namespace for these?
```suggestion
```

Reply:
Talked to @fspmarshall offline and we don't. I had mistakenly picked this up from another service, but all of the namespace stuff should be gone now :+1: